nixos/lib/utils.nix

```diff
@@ -8,4 +8,10 @@ rec {
    replaceChars ["/" "-" " "] ["-" "\\x2d" "\\x20"]
     (if hasPrefix "/" s then substring 1 (stringLength s) s else s);
 
+  # Returns a system path for a given shell package
+  toShellPath = shell:
+    if types.shellPackage.check shell then
+      "/run/current-system/sw${shell.shellPath}"
```

Comment:
I don't mind changing it but I don't know the side effects (huhu). Maybe @edolstra or @vcunat know why it was like this in the first place. Maybe the reason is just that it's hard to know the derivation path without having access to the derivation in the first place.